### PR TITLE
Task 7: add delete student feature

### PR DESCRIPTION
**Pull request by karlcloudy**
## Description
Implementation of task 7, which allows a student to be deleted.

## Breakdown of Commits
Commit 1 : 4d0c14f
- Add spec for Student destroy action.
- Implement destroy action for Student.
- Add 'Delete' button to Student list page.
 
Commit 2 : 346219f
- Add 'Delete' button to Student details page.

#### TODO Checklist
* [X] [Keep Your Pull Requests Small] (https://jobready.atlassian.net/wiki/spaces/DEV/pages/806551652/Keep+Your+Pull+Requests+Small)
* [X] [Code Review Process and Checklist](https://jobready.atlassian.net/wiki/spaces/DEV/pages/10256471/Pull+Requests+and+the+Code+Review+Process)
* [X] Add a Delete button / function to the Students list page.
  
#### Screenshot (If applicable)
image
image

### CODE REVIEW

File: spec/controllers/students_controller_spec.rb [hunk 1]
```diff
@@ -60,4 +60,17 @@
...
+  describe 'DELETE #destroy' do
+    subject { delete :destroy, id: student }
+
+    it 'redirects to the student list upon deletion' do
+      is_expected.to redirect_to students_path
+    end
+
+    it 'deletes the expected student' do
+      subject
+      expect { Student.find(student.id) }.to raise_error(ActiveRecord::RecordNotFound)
```
Comment (jhuizy) [inline spec/controllers/students_controller_spec.rb:73]: You can use the `.reload` method to let activerecord reload the student from the database, (ie. `student.reload`). You can also use the `be` matcher to check that the record still exists. Combining the two, we end up with something like 

 ```suggestion
      subject
      expect(student.reload).to_not be
```

which IMO reads a bit nicer, what do you think ?

**Review by jhuizy — COMMENTED** [review]
Just a cosmetic change to the spec, otherwise looks good 👍

File: spec/controllers/students_controller_spec.rb (hunk 1 above)
Comment (karlcloudy) [reply spec/controllers/students_controller_spec.rb]: It does, although `student.reload` also seems to raise `RecordNotFound`, so instead I have replaced it the following:
```rb
expect { student.reload }.to raise_error(ActiveRecord::RecordNotFound)
```

Comment (jhuizy) [reply spec/controllers/students_controller_spec.rb]: Yep that'll do 👍

**Review by jhuizy — APPROVED** [review]

**Pull request merged**